src/main/java/nu/studer/gradle/buildscan/teamcity/TeamCityBuildScanPlugin.java

```diff
@@ -22,25 +22,23 @@ public void apply(Project project) {
         if (System.getenv(TEAMCITY_VERSION_ENV) != null && System.getenv(GRADLE_BUILDSCAN_TEAMCITY_PLUGIN_ENV) == null) {
             project.getLogger().quiet(ServiceMessage.of(BUILD_SCAN_SERVICE_MESSAGE_NAME, BUILD_SCAN_SERVICE_STARTED_MESSAGE_ARGUMENT).toString());
 
-            project.getPluginManager().withPlugin(BUILD_SCAN_PLUGIN_ID, appliedPlugin -> {
-                BuildScanExtension buildScanExtension = project.getExtensions().getByType(BuildScanExtension.class);
-                if (supportsScanPublishedListener(buildScanExtension)) {
-                    buildScanExtension.buildScanPublished(publishedBuildScan -> {
+            BuildScanExtension buildScanExtension = project.getExtensions().findByType(BuildScanExtension.class);
+            if (supportsScanPublishedListener(buildScanExtension)) {
+                buildScanExtension.buildScanPublished(publishedBuildScan -> {
                             ServiceMessage serviceMessage = ServiceMessage.of(
-                                BUILD_SCAN_SERVICE_MESSAGE_NAME,
-                                BUILD_SCAN_SERVICE_URL_MESSAGE_ARGUMENT_PREFIX + publishedBuildScan.getBuildScanUri().toString()
+                                    BUILD_SCAN_SERVICE_MESSAGE_NAME,
+                                    BUILD_SCAN_SERVICE_URL_MESSAGE_ARGUMENT_PREFIX + publishedBuildScan.getBuildScanUri().toString()
                             );
                             project.getLogger().quiet(serviceMessage.toString());
                         }
-                    );
-                }
-            });
+                );
+            }
         }
     }
 
-    private static boolean supportsScanPublishedListener(BuildScanExtension extension) {
-        Class clazz = extension.getClass();
-        for (Method method : clazz.getMethods()) {
+    private static boolean supportsScanPublishedListener(@Nullable BuildScanExtension extension) {
+        if (extension == null) return false;
```

Comment:
Is there a test case that triggers this code path?

Reply:
Code path not there anymore.